### PR TITLE
DSD-197: Chakra CheckboxGroup

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-197](https://jira.nypl.org/browse/DSD-197)

## This PR does the following:
- Composes the `CheckboxGroup` with Chakra components.
- Makes minor modifications to the `RadioGroup` component.

## How has this been tested?

Locally, Library Card app, and Storybook.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr686-w0tpd5ppryl2hpw0dqndifar8xods0u9.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6149f021b8fde54327b3115e

File: src/components/CheckboxGroup/CheckboxGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  optReqFlag?: boolean;
+  /** Offers the ability to show the group's legend onscreen or hide it. Refer
+   * to the `labelText` property for more information. */
+  showLabel?: boolean;
+}
+
+const noop = () => {};
+
+/**
+ * Wrapper component to wrap `Checkbox` components. Can be displayed in a
```
Comment (bigfishdesign13) [inline src/components/CheckboxGroup/CheckboxGroup.tsx:52]: This is the description that is showing at the top of the story `Docs` screen.  Why use this and not something written right into the story file?

Wherever the final description sits, it should include details similar to what was included in the `Select` description.

```
The `CheckboxGroup` component renders all the necessary wrapping and associated text elements, but the checkbox input elements _need_ to be child `Checkbox` components from the NYPL Design System.`

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Looks very good.  One suggested change to the description, but also a question about how that description is implemented.

File: src/components/CheckboxGroup/CheckboxGroup.tsx (hunk 1 above)
Comment (EdwinGuzman) [reply src/components/CheckboxGroup/CheckboxGroup.tsx:52]: I find it easier to reuse the dev comment description for a component in Storybook using `<Description of={CheckboxGroup} />`. I updated the comment to include the above description.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr686-w0tpd5ppryl2hpw0dqndifar8xods0u9.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6149f021b8fde54327b3115e

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.

**Pull request merged**